### PR TITLE
Better queue config warning.

### PR DESCRIPTION
**Pull request by hjoliver**
This is a small change with no associated Issue.  Result of a user complaint today.

The following queue validation warning applies if the task `bar` (assigned to a queue) is not defined under `[runtime]` **or** if it is defined there but is simply not used in the graph:
```
WARNING - Queue configuration warnings:
        + q1: ignoring bar (task not defined)
```

On this branch, the warning in the latter case changes to: 
```
WARNING - Queue configuration warnings:
        + q1: ignoring bar (task not used in the graph)
```

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).

- [x] Appropriate tests are included (unit and/or functional).

- [x] Appropriate change log entry included.

- [x] No documentation update required.

### CODE REVIEW

**Comment by hjoliver** [comment]
This trivial change really demands a simple unit test, except that it requires the whole monolithic suite config class so at the moment a functional test would be much easier to write :grimacing:  Might be best to wait on @oliver-sanders latest experiment? #3616 (which I have seen but not read yet...).

**Comment by hjoliver** [comment]
That's weird, style fail in `cylc/flow/cfgspec/globalcfg.py` not touched by this branch :thinking:

**Comment by kinow** [comment]
@hjoliver I merged a PR that had `pycodestyle` issues I think, sorry. I thought it failed because of pending merge of other PR's in cylc-doc/cylc-sphinx-extensions. 

https://github.com/cylc/cylc-flow/pull/3559/checks?check_run_id=685749339

**Comment by oliver-sanders** [comment]
This is definitely a unit-test problem rather than an integration-test.

Unit-testing this isn't especially easy due to complex objects getting in the way, it would be a lot easier if the code was functional, however, it is still doable:

Create a `Mock` object, then have a quick scan down the code searching for `self.` to see what you have to bolt onto it to get to the required branch in the code. You usually only need a few things:

```python
# untested

def test_queue_thinggy(caplog):
    config = Mock()
    config.Q_DEFAULT = 'default'
    config.cfg = {
        'scheduling': {
            'queues': {
                'default': {}
            }
        },
        'runtime': {}
    }
    config.runtime = {
        'descendents': []
    }

    caplog.set_level(logging.INFO, logger='cylc')
    Config.configure_queues(config)
    assert caplog.reccord_tuples = [
        # expected log messages
    ]
```

**Comment by hjoliver** [comment]
Ha, there's already a `test_queue_config` function in `tests/test_config.py` (using `caplog` too) ... and what's more, `git blame` says I wrote it.  I think I need my brain looked at :rofl:  It's not quite as clean as  your suggestion as it uses a predefined SuiteConfig object (with a bunch of mocked bits inside) but it'll do.

**Comment by hjoliver** [comment]
Rebased to pick up the off-topic style fix; unit test added.

File: cylc/flow/tests/test_config.py [hunk 1]
```diff
@@ -236,3 +237,30 @@ def test_queue_config(caplog, tmp_path):
...
+    suiterc_content = """
+[scheduling]
+   [[queues]]
+       [[[q1]]]
+           members = foo
+       [[[q2]]]
+           members = bar
+   [[dependencies]]
+       # foo and bar not used
+       graph = "beef => wellington"
```
Comment (oliver-sanders) [inline cylc/flow/tests/test_config.py:253]: 🍌

**Pull request merged**